docsp-31169 - change stream schema inference (#164)
mongodb/docs-spark-connector: source/read-from-mongodb.txt

         .. include:: /scala/filters.txt

.. important:: Change Stream Schema Inference

>    When the {+driver-short+} infers the schema of a data frame 

Reviewer: Q: should we rename this source constant since the product is not a driver?

Author: love it

   read from a change stream, by default,

>    it will use the schema of the underlying collection rather than the schema

Reviewer: S: Remove the "by default" here. It is implicit as the next sentence explains how to change the behavior

Author: I see what you mean, but to me it benefits from some indication that this behavior can be changed (apart from the following sentence, I mean). Especially when this isn't the same page where this option is documented, so the reader might not be in the mindset of settings they can turn on or off.

>    When the {+driver-short+} infers the schema of a data frame 
>    read from a change stream, by default,
>    it will use the schema of the underlying collection rather than the schema
>    of the change stream. To instruct the connector to use the schema of the 

Reviewer: ```suggestion
   When the {+driver-short+} infers the schema of a data frame 
   read from a change stream, by default,
   it will use the schema of the underlying collection rather than of the change stream.
   To instruct the connector to use the schema of the 
```

Author: Used your suggestion plus "that" to hopefully make it more clear. LMK

>    of the change stream. To instruct the connector to use the schema of the 
>    change stream, set the ``change.stream.publish.full.document.only`` option 
>    to ``true``.

Reviewer: ```suggestion
   of the change stream. The connector infers the schema from the 
   change stream if you set the ``change.stream.publish.full.document.only`` option 
   to ``true``.
```

Author: reworded. just for my info, what don't you like about the first version?

Reviewer: The phrasing of "instruct the connector" seemed needlessly complicated. Better to just have it be a declarative sentence, IMO. but it just a suggestion :)

   For more information on configuring a read operation, see the

>    :ref:`spark-read-conf` guide.

Reviewer: S: make this link more specific about the change read read options and link here https://docs-mongodbcom-staging.corp.mongodb.com/spark-connector/docsworker-xlarge/docsp-31169-infer-schema/configuration/read/#change-streams

Author: that's what i had initially. not sure why i changed
